types/defines/trace.d.ts

```diff
@@ -188,9 +189,29 @@ type EventType =
   | Link
   | Attributes;
 
+// Context in which this trace event lives.
+interface SpanContext {
+  // Single id for the entire top-level invocation
+  // This should be a new traceId for the first worker stage invoked in the eyeball request and then
+  // same-account service-bindings should reuse the same traceId but cross-account service-bindings
+  // should use a new traceId.
+  readonly traceId: string;
+  // spanId in which this event is handled
+  // for Onset and SpanOpen events this would be the parent span id
+  // for Outcome and SpanClose these this would be the span id of the opening Onset and SpanOpen events
+  // For Hibernate and Mark this would be the span under which they were emitted.
+  // spanId is not set ONLY if:
```

Comment:
- [ ] Check that this condition is implemented properly – spanId might be incorrectly present even in this one special case, and we're providing the InvocationSpanContext spanId as this for non-span events even though the Onset itself has all-zeroes as its inner spanId.
- [ ] These are great candidates to be included in tests – previously I was unsure how to test the outer event in a deterministic test, but we certainly can check if these conditions are handled correctly.